### PR TITLE
Feat: onboarding data privacy

### PR DESCRIPTION
**Pull request by cimigree**
This is the two screens that follow the first screen when onboarding.
related to #42 
They are mentioned in that Issue but the UI isn't described.
It can be found in Figma:
https://www.figma.com/design/UIA00RSP8Sz24V0vXvQHeL/(Desktop)-CoMapeo?node-id=0-1&node-type=canvas&t=4esd3jysAvxz8VlD-0
I had to add a bunch of images, some of which are SVG. If I could, I used pngs.
Right now I am just mocking the metrics toggle because the Logic from metrics that exists in mobile is not in the desktop app [yet?].

**If you click on Get Started, you get to this screen:**

Screenshot 2024-12-02 at 6 28 31 PM

**And then if you click on Learn More, you get to this screen:**

Screenshot 2024-12-02 at 6 29 09 PM

**That is a very, very long scrollable screen.**

Screenshot 2024-12-02 at 6 30 35 PM
Screenshot 2024-12-02 at 6 30 54 PM
Screenshot 2024-12-02 at 6 31 08 PM
Screenshot 2024-12-02 at 6 31 22 PM

**The Go back button will take you back to the previous screen. This is all Step 1 and that stays highlighted throughout **

### CODE REVIEW

File: messages/renderer/en.json
Comment (EvanHahn) [inline messages/renderer/en.json:1]: I didn't review any of the translation strings in this file or others. Let me know if that's wrong.

File: src/renderer/src/components/OnboardingTopMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+			<GoBackButton
+				onClick={() => router.history.back()}
+				variant="text"
+				style={{
+					color: BLUE_GREY,
+					gap: 8,
+					padding: '12px 32px',
+				}}
+			>
+				<BackArrow>←</BackArrow>
```
Comment (EvanHahn) [inline src/renderer/src/components/OnboardingTopMenu.tsx:92]: I'd love to give this an `aria-label` for screen readers. I don't think screen readers will do a great job with this.

File: src/renderer/src/components/OnboardingTopMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+												: 'PrivacyPolicyScreen'
+									}`,
+								})
+							}
+						>
+							<Text kind="body" bold={currentStep === step}>
+								{formatMessage(m.step, { number: step })}
+							</Text>
+						</Step>
+						{step < 3 && <Divider />}
```
Comment (EvanHahn) [inline src/renderer/src/components/OnboardingTopMenu.tsx:116]: I think this is fine, but could we do this with CSS? (Feel free to ignore.)

File: src/renderer/src/components/OnboardingTopMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+						<Step
+							active={currentStep === step}
+							onClick={() =>
+								navigate({
+									to: `/Onboarding/${
+										step === 1
+											? 'DataPrivacy'
+											: step === 2
+												? 'NextStep'
+												: 'PrivacyPolicyScreen'
```
Comment (EvanHahn) [inline src/renderer/src/components/OnboardingTopMenu.tsx:107]: Nit: might just be me, but I find nested ternary expressions hard to read. Maybe something like this, if you want to maintain this being a single expression?

```suggestion
										{
											1: 'DataPrivacy',
											2: 'NextStep',
											3: 'PrivacyPolicyScreen',
										}[step]
```

File: src/renderer/src/components/PrivacyPolicy/MetricsDiagnosticsPermissionToggle.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+		const savedValue = localStorage.getItem('MetricDiagnosticsPermission')
+		return savedValue !== null ? savedValue === 'true' : true
+	})
+
+	const togglePermission = () => {
+		const newValue = !isEnabled
+		setIsEnabled(newValue)
+		localStorage.setItem('MetricDiagnosticsPermission', String(newValue))
+
+		console.log('Permission updated:', newValue)
```
Comment (EvanHahn) [inline src/renderer/src/components/PrivacyPolicy/MetricsDiagnosticsPermissionToggle.tsx:38]: Nit: maybe add a TODO to indicate that this code is going to be replaced?

File: src/renderer/src/components/PrivacyPolicy/PointContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	description: string
+}) {
+	const isSvg = typeof icon !== 'string'
+	return (
+		<Container>
+			<PointHeader>
+				{isSvg && icon ? (
+					React.createElement(icon, { width: 18, height: 18 })
+				) : (
+					<img src={icon as string} alt="" width={18} height={18} />
```
Comment (EvanHahn) [inline src/renderer/src/components/PrivacyPolicy/PointContainer.tsx:41]: Nit: I think TypeScript is smart enough to know that `icon` is a string, so we don't need this cast.

```suggestion
					<img src={icon} alt="" width={18} height={18} />
```

File: src/renderer/src/images/RedDot.svg
Comment (EvanHahn) [inline src/renderer/src/images/RedDot.svg:1]: The gradient on this dot looks a little weird. Not sure if that's intentional...feel free to ignore if so.

**Review by EvanHahn — APPROVED** [review]
LGTM other than a few nitpicks.

File: messages/renderer/en.json
Comment (cimigree) [reply messages/renderer/en.json:1]: I think that is fine!

File: src/renderer/src/components/OnboardingTopMenu.tsx (hunk 1 above)
Comment (cimigree) [reply src/renderer/src/components/OnboardingTopMenu.tsx:92]: Ok done

File: src/renderer/src/components/OnboardingTopMenu.tsx (hunk 2 above)
Comment (cimigree) [reply src/renderer/src/components/OnboardingTopMenu.tsx:116]: Sorry @EvanHahn I am not sure what you mean. Do what exactly with CSS?

File: src/renderer/src/images/RedDot.svg
Comment (cimigree) [reply src/renderer/src/images/RedDot.svg:1]: I see what you mean. I found a new one to use.

File: src/renderer/src/components/OnboardingTopMenu.tsx (hunk 3 above)
Comment (cimigree) [reply src/renderer/src/components/OnboardingTopMenu.tsx]: Good point! I replaced it with an object as you suggested.

File: src/renderer/src/components/OnboardingTopMenu.tsx (hunk 2 above)
Comment (cimigree) [reply src/renderer/src/components/OnboardingTopMenu.tsx:113]: I am going to merge this PR in for now, but I can go back and fix later once I understand.

Comment (EvanHahn) [reply src/renderer/src/components/OnboardingTopMenu.tsx:113]: Not important. I think what you have here is totally fine.

**Pull request merged**